packages/grid/_modules_/grid/components/column-header-filter-icon.tsx

```diff
@@ -0,0 +1,232 @@
+        });
+      },
+      [apiRef, column],
+    );
+
+    return (
+      <div className={'MuiDataGrid-iconFilter'}>
+        <IconButton aria-label="Sort" size="small" onClick={filterClick}>
+          {icon}
+        </IconButton>
+      </div>
+    );
+  },
+);
+ColumnHeaderFilterIcon.displayName = 'ColumnHeaderFilterIcon';
+
+function TabPanel(props) {
+  const { children, value, index, ...other } = props;
+
+  return (
+    <div
+      role="tabpanel"
+      hidden={value !== index}
+      id={`simple-tabpanel-${index}`}
+      aria-labelledby={`simple-tab-${index}`}
+      {...other}
+    >
+      {value === index && (
+        <Box p={3}>
+          <Typography>{children}</Typography>
+        </Box>
+      )}
+    </div>
+  );
+}
+
+// const useStyles = makeStyles((theme) => ({
+//   paper: {
+//     maxWidth: 400,
+//     overflow: 'auto',
+//   },
+//   popper: {
+//     zIndex: 1,
+//     '&[x-placement*="bottom"] $arrow': {
+//       top: 0,
+//       left: 0,
+//       marginTop: '-0.9em',
+//       width: '3em',
+//       height: '1em',
+//       '&::before': {
+//         borderWidth: '0 1em 1em 1em',
+//         borderColor: `transparent transparent ${theme.palette.background.paper} transparent`,
+//       },
+//     },
+//     '&[x-placement*="top"] $arrow': {
+//       bottom: 0,
+//       left: 0,
+//       marginBottom: '-0.9em',
+//       width: '3em',
+//       height: '1em',
+//       '&::before': {
+//         borderWidth: '1em 1em 0 1em',
+//         borderColor: `${theme.palette.background.paper} transparent transparent transparent`,
+//       },
+//     },
+//     '&[x-placement*="right"] $arrow': {
+//       left: 0,
+//       marginLeft: '-0.9em',
+//       height: '3em',
+//       width: '1em',
+//       '&::before': {
+//         borderWidth: '1em 1em 1em 0',
+//         borderColor: `transparent ${theme.palette.background.paper} transparent transparent`,
+//       },
+//     },
+//     '&[x-placement*="left"] $arrow': {
+//       right: 0,
+//       marginRight: '-0.9em',
+//       height: '3em',
+//       width: '1em',
+//       '&::before': {
+//         borderWidth: '1em 0 1em 1em',
+//         borderColor: `transparent transparent transparent ${theme.palette.background.paper}`,
+//       },
+//     },
+//   },
+//   arrow: {
+//     position: 'absolute',
+//     fontSize: 7,
+//     width: '3em',
+//     height: '3em',
+//     '&::before': {
+//       content: '""',
+//       margin: 'auto',
+//       display: 'block',
+//       width: 0,
+//       height: 0,
+//       borderStyle: 'solid',
+//     },
+//   },
+// }));
+
+export const ColumnFilterMenu: React.FC = () => {
+  // const classes = useStyles();
+  const apiRef = React.useContext(ApiContext);
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [filterValue, setFilterValue] = React.useState('');
+  const [target, setTarget] = React.useState<HTMLElement | null>(null);
+  const [colDef, setColumn] = React.useState<ColDef | null>(null);
```

Comment:
For consistency:

```suggestion
  const [colDef, setColDef] = React.useState<ColDef | null>(null);
```

or

```suggestion
  const [column, setColumn] = React.useState<ColDef | null>(null);
```